### ISSUE TITLE
Thumbnails no longer load, allowed memory exceeded

### ISSUE THREAD

**Issue by emichael** [nextcloud/maps#805]
I'm on the latest version of Maps and a docker install of Nextcloud 24.0.3. Maps no longer functions. When I load the maps page, after a few seconds my photos appear in the correct spots, but the thumbnails are all solid white. Moreover, once the thumbnails appear, I can no longer load any more map tiles. I can pan and zoom, but new map tiles do not appear.

I see a bunch of these errors in the logs. I can't tell if they're from page loads or the cron job.

[PHP] Error: Allowed memory size of 1073741824 bytes exhausted (tried to allocate 5322080 bytes) at /var/www/html/custom_apps/maps/lib/Helper/ExifGeoData.php#120

**Comment by freaky33** [comment]
Really sad to see that apparently nobody seems to care about these types of issues anymore, you can see that with the main nextcloud project too.

I've got the same issues. I think it has to do with the cronjob or execution of it since it works perfectly fine if I completely disable the nextcloud cron job. But after reenabling it it will become unresponsive again until the point where the whole nextcloud instance isn't reachable anymore. 

I can see that there are multiple php-fpm processes being created and they start to utilize the CPU up to a 100% until they disappear again after a while.

**Comment by tacruc** [comment]
> Really sad to see that apparently nobody seems to care about these types of issues anymore, you can see that with the main nextcloud project too.

I'm sorry that you feel this way. But apparently, you care! So why don't you start to fix them? If you can not, can you explain why? 
Can you ask for help or support enabling you to fix them? 

> I've got the same issues

That's interesting, I have a hard time understanding what is the "Issue" here. 

I See many:

- We still have no Issue template enabling people to provide the  information required to reproduce the Issue.
- There is a problem in the process either in Thumbnails 
  - generation (on the server)
  - loading (request and transfer from server to browser)
  - showing (they are correctly loaded in the browser but not shown)
-  _Moreover, once the thumbnails appear_ (Wait I thought they are just white?) so everything is working just slow?
Anyhow apparently then something else breaks. 
  - The log from the Browser console would be interesting then. 
  - The Network Monitor can additionally provide information on how long the loading of the thumbnails from the server takes.
- Then we have the error from the log, which has nothing to do with the not working frontend/ javascript part. Coming from the cron job. There are again some open questions 
  - The most prominent one, is what is executed in the cron job? 
  `cron.php` or `occ maps:scan-photos` 

In conclusion, I don't know exactly what the issue and I don't know how to reproduce it. I don't know how to show that I care about it. And I still would like to get it fixed.
So @freaky33 what are your suggestions? How shall we continue with this Issue?

**Comment by tacruc** [comment]
> Well, being snarky
> 
>     (Wait I thought they are just white?)
> 
> certainly isn't a good way to show that you care. I'm really not sure why my post is being dragged. If you want clarification, just ask for it.

I know and I'm sorry. I'm happy that you still took your time to answer the questions.

- Do the Preview Urls, load when you just load one preview? So is it some kind of performance problem and the maps app basically running a ddos on your server?

- Are previews in other apps loading fine?

In general information about preview generation are found:
https://docs.nextcloud.com/server/latest/admin_manual/configuration_files/previews_configuration.html
and 
https://apps.nextcloud.com/apps/previewgenerator
is quite helpful, too.

**Comment by emichael** [comment]
Previews seem to be working for other apps (e.g., Photos) just fine, until I load up the maps and hit the above issue. After that, previews will not load. When I manually open a single preview link that maps is requesting, it eventually does load.

One other piece of info: running `htop` on the server while hitting the issue shows that CPU and memory usage are very low. I'm not running out of those resources.

**Comment by tacruc** [comment]
You could check out the v0.2.5 nightly builds.  It includes #924 , which brings improvements.
Now not all previews are loaded, but just one for each cluster.

Feel free to reopen if the problem still exits there.
